engine/crates/telemetry/src/gql_response_status.rs

```diff
@@ -90,3 +88,28 @@ impl headers::Header for GraphqlResponseStatus {
         values.extend(Some(GraphqlResponseStatus::encode(self).try_into().unwrap()))
     }
 }
+
+#[derive(Clone, Copy, Debug)]
+pub enum SubgraphResponseStatus {
```

Comment:
would rename `GraphqlRequestStatus` and re-use it for the top level gateway request span